docs/system_overview/genquery.md

```diff
@@ -553,3 +551,281 @@ Find all data objects with replicas on resource `replResc` which are not marked
+| RULE_EXEC_STATUS | DELAY_RULE_STATUS |
+| RULE_EXEC_TIME | DELAY_RULE_EXE_TIME |
+| RULE_EXEC_USER_NAME | DELAY_RULE_USER_NAME |
+
+As mentioned earlier, GenQuery2 exposes a subset of the GenQuery1 attributes. That means, many of the attributes were dropped from GenQuery2. Notable attributes include the following:
+
+- `COLL_OWNER_NAME`
+- `COLL_OWNER_ZONE`
+- `DATA_OWNER_NAME`
+- `DATA_OWNER_ZONE`
+
+Logic relying on these attributes will need to be updated to use the permission attributes.
+
+To get the full listing of attributes, use `iquery -c`. The listing can also be retrieved programmatically via the `rc_genquery2` API endpoint.
+
+!!! Note
+    The results always reflect the attributes supported by the GenQuey2 implementation of the iRODS Catalog Provider.
+
+### Aggregation Operators and SQL functions
+
+Support for aggregation operators (and SQL functions) is significantly improved in GenQuery2.
+
+All aggregation operators supported by GenQuery1 are supported by GenQuery2. However, GenQuery2 extends support for aggregation operators by allowing the following:
+
+- Aggregation operators can be nested
+- Aggregation operators can accept any number of input arguments
+- SQL functions can be used in the PROJECTION, WHERE, and ORDER-BY clauses
+
+Below are a few examples demonstrating each feature.
+
+```sh
+# SQL functions in the PROJECTION clause.
+iquery "select concat(COLL_NAME, concat('--', COLL_ACCESS_USER_NAME)) where COLL_NAME = '/tempZone/home/rods'"
+
+# SQL functions in the WHERE clause.
+iquery "select COLL_NAME where concat(COLL_NAME, concat('--', COLL_ACCESS_USER_NAME)) = '/tempZone/home/rods--rods'"
+
+# SQL functions in the ORDER-BY clause. 
+iquery "select concat(COLL_NAME, concat('--', COLL_ACCESS_USER_NAME)) where COLL_NAME = '/tempZone/home/rods' order by concat(COLL_NAME, concat('--', COLL_ACCESS_USER_NAME))"
+
+# Case-insensitive search.
+iquery "select DATA_NAME where lower(DATA_NAME) = 'foo'"
+iquery "select DATA_NAME where upper(DATA_NAME) = 'FOO'"
+
+# Sub-string and integers.
+iquery "select substr(DATA_NAME, 1, 3)"
+```
+
+!!! Note
+    The behavior of aggregation operators and SQL functions depends on the database. GenQuery2 only acts as a passthrough.
+
+### Metadata Queries
+
+GenQuery2 allows users to write more targeted queries. This is especially helpful when dealing with metadata. Depending on the use-case, there may be times where you need to filter a resultset based on metadata attached to multiple iRODS entities (i.e. data objects, collections, resources, users). GenQuery2 makes this possible through the use of **SQL LEFT JOIN**. This happens automatically.
+
+Here's an example demonstrating mixed metadata queries. _This is NOT possible with GenQuery1._
+
+```sh
+$ itouch foo
+$ imeta add -d foo id 1000
+$ itouch -R otherResc bar
+$ imeta add -R otherResc speed fast
+$ iquery "select DATA_NAME where META_DATA_ATTR_VALUE = '1000' or META_RESC_ATTR_VALUE = 'fast'" | jq
```

Comment:
A comment explaining what this query does may be helpful. Is it selecting only replicas which reside on resources with that metadata?

Reply:
It returns the data names which satisfy either of the following:
- There exists a data object with an attribute value of 1000 attached to it
- Or, a replica of a data object exists on a resource which has an attribute value of 'fast' attached to the resource

Both data objects are returned because both of them satisfy exactly one of the conditions.

Will add some words about that.

Reply:
Done.